embassy-stm32/src/spi/v1.rs

```diff
@@ -128,9 +141,150 @@ impl<'d, T: Instance> Spi<'d, T> {
             self.current_word_size = word_size;
         }
     }
+
+    #[allow(unused)]
+    async fn write_dma_u8(&mut self, write: &[u8]) -> Result<(), Error>
+    where
+        Tx: TxDmaChannel<T>,
+    {
+        unsafe {
+            T::regs().cr1().modify(|w| {
+                w.set_spe(false);
+            });
+            T::regs().cr2().modify(|reg| {
+                reg.set_rxdmaen(true);
+            });
+        }
+        self.set_word_size(WordSize::EightBit);
+
+        let request = self.txdma.request();
+        let dst = T::regs().dr().ptr() as *mut u8;
+        let f = self.txdma.write(request, write, dst);
+
+        unsafe {
+            T::regs().cr2().modify(|reg| {
+                reg.set_txdmaen(true);
+            });
+            T::regs().cr1().modify(|w| {
+                w.set_spe(true);
+            });
+        }
+
+        f.await;
+        Ok(())
+    }
+
+    #[allow(unused)]
+    async fn read_dma_u8(&mut self, read: &mut [u8]) -> Result<(), Error>
+    where
+        Tx: TxDmaChannel<T>,
+        Rx: RxDmaChannel<T>,
+    {
+        unsafe {
+            T::regs().cr1().modify(|w| {
+                w.set_spe(false);
+            });
+            T::regs().cr2().modify(|reg| {
+                reg.set_rxdmaen(true);
+            });
+        }
+        self.set_word_size(WordSize::EightBit);
+
+        let clock_byte_count = read.len();
+
+        let rx_request = self.rxdma.request();
+        let rx_src = T::regs().dr().ptr() as *mut u8;
+        let rx_f = self.rxdma.read(rx_request, rx_src, read);
+
+        let tx_request = self.txdma.request();
+        let tx_dst = T::regs().dr().ptr() as *mut u8;
+        let clock_byte = 0x00;
+        let tx_f = self
+            .txdma
+            .write_x(tx_request, &clock_byte, clock_byte_count, tx_dst);
+
+        unsafe {
+            T::regs().cr2().modify(|reg| {
+                reg.set_txdmaen(true);
+            });
+            T::regs().cr1().modify(|w| {
+                w.set_spe(true);
+            });
+        }
+
+        join3(tx_f, rx_f, Self::wait_for_idle()).await;
+
+        unsafe {
+            T::regs().cr2().modify(|reg| {
+                reg.set_txdmaen(false);
+                reg.set_rxdmaen(false);
+            });
+            T::regs().cr1().modify(|w| {
+                w.set_spe(false);
+            });
+        }
+
+        Ok(())
+    }
+
+    #[allow(unused)]
+    async fn read_write_dma_u8(&mut self, read: &mut [u8], write: &[u8]) -> Result<(), Error>
```

Comment:
Should we check for the buffers being the same size ? What happens if `read.len() > write.len()` ?

Reply:
Added both an assert that read >= write, slicing it down as needed, and documented.